File: sorting/selectionSort.js
const swap = (array, i, j) => {
  const temp = array[i];
  array[i] = array[j];
  array[j] = temp;
};

// Time complexity: O(n^2)
// Space complexity: O(1)
const selectionSort = (array) => {
  for (let i = 0; i < array.length; i++) {
    let minIndex = i;
    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < array[minIndex]) minIndex = j;
    }
    if (i !== minIndex) swap(array, i, minIndex);
  }
  return array;
};

File: sorting/selectionSort.py
from typing import List

def selectionSort(nums: List[int]) -> List[int]:
  n = len(nums)
  for i in range(n):
    minIndex = i
    for j in range(i + 1, n):
      if nums[j] < nums[minIndex]: minIndex = j
    if i != minIndex: nums[i], nums[minIndex] = nums[minIndex], nums[i]
  return nums 
print(selectionSort([3, 2, 1, 5, 6, 4]))
